File: javascript/src/models/event_type_update.ts
// this file is @generated

export interface EventTypeUpdate {
  archived?: boolean;
  deprecated?: boolean;
  description: string;
  featureFlag?: string | null;
  /** The event type group's name */
  groupName?: string | null;
  /** The schema for the event type for a specific version as a JSON schema. */
  schemas?: any | null;
}

export namespace EventTypeUpdateUtil {
  export function _fromJsonObject(object: any): EventTypeUpdate {
    return {
      archived: object["archived"],
      deprecated: object["deprecated"],
      description: object["description"],
      featureFlag: object["featureFlag"],
      groupName: object["groupName"],
      schemas: object["schemas"],
    };
  }

  export function _toJsonObject(self: EventTypeUpdate): any {
    return {
      archived: self.archived,
      deprecated: self.deprecated,
      description: self.description,
      featureFlag: self.featureFlag,
      groupName: self.groupName,
      schemas: self.schemas,
    };
  }
}

File: kotlin/lib/src/main/kotlin/models/EventTypeUpdate.kt
// This file is @generated
package com.svix.kotlin.models

import kotlinx.serialization.Serializable
import kotlinx.serialization.json.JsonObject

@Serializable
data class EventTypeUpdate(
    val archived: Boolean? = null,
    val deprecated: Boolean? = null,
    val description: String,
    val featureFlag: String? = null,
    val groupName: String? = null,
    val schemas: JsonObject? = null,
)
